test/dump/invalid-data-segment-no-memory.txt

```diff
@@ -1,3 +1,4 @@
+;;; TOOL: wat2wasm
```

Comment:
So part of this change to remove the default tool?  Perhaps that could be a separate CL since it seems somewhat orthogonal to supporting multiple tools.

Reply:
No, it still has a default tool, but it doesn't work with just `FLAGS`. I guess it would be better to just set it as the default if the `FLAGS` directive is found.